torch/_inductor/scheduler.py

```diff
@@ -2189,6 +2185,23 @@ def can_fuse_vertical(self, node1, node2):
         # we still can match unmet dep
         # if there's indirect indexing, don't match it
         def fusable_read_and_write(read: Dep, write: Dep):
+            read_name = self.mutation_renames.get(read.name, read.name)
+            write_name = self.mutation_renames.get(write.name, write.name)
+            if (
+                isinstance(read, MemoryDep)
+                and isinstance(write, MemoryDep)
+                and read.mode == write.mode
+                and write.mode is not None
+            ):
+                return True
+            if (
+                isinstance(read, StarDep)
+                and isinstance(write, MemoryDep)
+                and read.mode == write.mode
+                and write.mode is not None
+                and read_name == write_name
```

Comment:
Do we have any tests where we do something like:
```python
x.index_add_(...)
y = x.clone()
x.index_add_(...)
return x.clone(), y
```

and see that the two `index_add_` operations don't get fused into the same kernel?